### ISSUE TITLE
Preventing node-gyp rebuild on every cf push

### ISSUE THREAD

**Issue by charyorde** [cloudfoundry/nodejs-buildpack#31]
Hi,

Some of my app deps require a rebuild of native modules as seen below:
```
-------> Buildpack version 1.5.0
-----> Creating runtime environment
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
       NODE_MODULES_CACHE=true
-----> Installing binaries
       engines.node (package.json):  0.12.5
       engines.npm (package.json):   2.11.2
       Downloading and installing node 0.12.5...
       npm 2.11.2 already installed with node
-----> Restoring cache
       Loading 1 from cacheDirectories (default):
       - node_modules (exists - skipping)
-----> Building dependencies
       Prebuild detected (node_modules already exists)
       Rebuilding any native modules
       > fsevents@0.3.5 install /tmp/staged/app/node_modules/pm2/node_modules/chokidar/node_modules/fsevents
       > node-gyp rebuild
```
Is there any exposed flag to prevent this on every `cf push`?

**Comment by charyorde** [comment]
Yes I'm experiencing problems like app taking too long to stage. I had to increase timeout using `cf push -t 180`. With that said, during app health check (like one of the instances fails and tries to restart itself), it leaves the app in unstable state. Feels like it reduces the app performance.

**Comment by jtarchie** [comment]
To your original question, there is no exposed flag. You are free to fork the buildpack and add the support yourself.

If you are having app performance problems, please contact your CF provider for support.

**Comment by flavorjones** [comment]
@charyorde Worth noting is that we received PR #19 for something like what you're suggesting.

IMHO, it shouldn't be something that a user has to set, the buildpack (or maybe node-gyp) should be smart enough to see if artifacts are already compiled in the cache dir, and re-use them if so.

We're currently blocked on moving forward because we lack the expertise to know how to implement such a feature, and we're looking for someone from the node.js community who'd be willing to pair with us on this. Would you be interested in doing so?

**Comment by charyorde** [comment]
Seems the newly released NodeJS v4.0.0 solves this problem. 

Quoting from https://nodejs.org/en/blog/release/v4.0.0/ 
> node-gyp: This release bundles an updated version of node-gyp that works with all versions of Node.js and io.js including nightly and release candidate builds. From io.js v3 and Node.js v4 onward, it will only download a headers tarball when building addons rather than the entire source. (Rod Vagg) https://github.com/nodejs/node/pull/2700
